hooks/hookexecution/execution.go

```diff
@@ -0,0 +1,234 @@
+package hookexecution
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/prebid/prebid-server/hooks"
+	"github.com/prebid/prebid-server/hooks/hookstage"
+)
+
+type TimeoutError struct{}
+
+func (e TimeoutError) Error() string {
+	return fmt.Sprint("Hook execution timeout")
+}
+
+// FailureError indicates expected error occurred during hook execution on the module-side
+type FailureError struct{}
+
+func (e FailureError) Error() string {
+	return fmt.Sprint("Hook execution failed")
+}
+
+type RejectError struct {
+	Code   int
+	Reason string // is it needed or code is enough?
+}
+
+func (e RejectError) Error() string {
+	return fmt.Sprintf(`Module rejected stage, reason: "%s"`, e.Reason)
+}
```

Comment:
Nitpick: can we move these errors into an `errors.go` file in this package?

Reply:
Sure